### PR TITLE
Change pipeline gpdb_src point to main branch

### PR DESCRIPTION
**Pull request by BaiShaoqi**
Since greenplum-db/gpdb has changed rename branch master to main

[GPR-1075]

Co-authored-by: Lucas Bonner <blucas@vmware.com>
Co-authored-by: Shaoqi Bai <bshaoqi@vmware.com>

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by lucasbonner — COMMENTED** [review]
Looks good!

**Review by lucasbonner — APPROVED** [review]

**Comment by khuddlefish** [comment]
We're going to want to backport the gen_pipeline.py changes to 6x and 5x. It might make sense to split this into two commits so you can backport one part that will apply to all 3 branches.

**Comment by BaiShaoqi** [comment]
with @lucasbonner  Thanks for your suggestion.
We notice there are 3 changes in gpdb main branch for gen_pipeline.py , while there are 2 changes in gpdb 6x_stable, and 1 change in gpdb 5X-stable, so the commit can not be cherry-picked to those the 6x_stable and 5X-stable branch.

File: concourse/pipelines/README.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ fly -t prod \
     -l ~/workspace/gpdb/concourse/vars/common_prod.yml \
     -l ~/workspace/gpdb/concourse/vars/without_asserts_common_prod.yml \
     -v gpdb-git-remote=https://github.com/greenplum-db/gpdb.git \
-    -v gpdb-git-branch=master \
+    -v gpdb-git-branch=main \
     -v pipeline-name=gpdb_master_without_asserts
```
Comment (BaiShaoqi) [inline concourse/pipelines/README.md:104]: with @50wu , also change the readme of the pipeline-name

Comment (50wu) [reply concourse/pipelines/README.md]: @BaiShaoqi @lucasbonner  I make a fixup commit to update the left `master` to `main`. Feel free to rebase and merge if you feel ok.

File: concourse/pipelines/pr_pipeline.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ resources:
   type: pull-request
   source:
     repository: greenplum-db/gpdb
-    base_branch: "main"
+    base_branch: "master"
```
Comment (gpopt) [inline concourse/pipelines/pr_pipeline.yml:21]: This is not expected, right?

**Pull request merged**